cookbooks/aws-parallelcluster-config/resources/manage_efs.rb

```
@@ -14,16 +14,33 @@
 
 property :shared_dir_array, Array, required: true
 property :efs_fs_id_array, Array, required: true
+property :efs_encryption_in_transit_array, Array, required: true
+property :efs_iam_authorization_array, Array, required: true
 
 default_action :mount
 
 action :mount do
   efs_shared_dir_array = new_resource.shared_dir_array.dup
   efs_fs_id_array = new_resource.efs_fs_id_array.dup
-  efs_fs_id_array.zip(efs_shared_dir_array).each do |efs_fs_id, efs_shared_dir|
+  efs_encryption_in_transit_array = new_resource.efs_encryption_in_transit_array.dup
+  efs_iam_authorization_array = new_resource.efs_iam_authorization_array.dup
+
+  efs_fs_id_array.each_with_index do |efs_fs_id, index|
+    efs_shared_dir = efs_shared_dir_array[index]
+    efs_encryption_in_transit = efs_encryption_in_transit_array[index]
+    efs_iam_authorization = efs_iam_authorization_array[index]
+
     # Path needs to be fully qualified, for example "shared/temp" becomes "/shared/temp"
     efs_shared_dir = "/#{efs_shared_dir}" unless efs_shared_dir.start_with?('/')
 
+    mount_options = "defaults"
+    if efs_encryption_in_transit == "true" && efs_iam_authorization == "true"
+      mount_options = "tls,iam"
+    elsif efs_encryption_in_transit == "true"
+      mount_options = "tls"
+      # efs_iam_authorization cannot be true alone. efs_iam_authorization requires efs_encryption_in_transit
```

I assume that a validation of these parameters will be done at the CLI level.
Nevertheless, I think it's also better to handle unsupported configurations at the cookbook level.
To this aim, it'd be better to fail with explicative error message when unsupported values are provided rather than hiding that.

What if I submit a cluster creation overriding cookbook attributes via DevSettings with `efs_iam_authorization = true` and `efs_encryption_in_transit = false`? With the proposed approach you would hide the invalid configuration, but you should fail instead.

---

Done